search-commons/src/main/java/no/unit/nva/search2/constant/Resource.java

```diff
@@ -119,16 +125,73 @@ public final class Resource {
         + ENTITY_PUBLICATION_INSTANCE_DOT + "manifestations" + DOT + ID + DOT + KEYWORD + PIPE
         + ENTITY_PUBLICATION_INSTANCE_DOT + ID + DOT + KEYWORD;
 
+
     public static final List<AbstractAggregationBuilder<? extends AbstractAggregationBuilder<?>>>
         RESOURCES_AGGREGATIONS = List.of(
-        generateSimpleAggregation(TYPE, PUBLICATION_INSTANCE_TYPE),
-        generatePublisher(),
-        generateContributor(),
-        generateFundingSource(),
-        generateObjectLabelsAggregation(TOP_LEVEL_ORGANIZATION, TOP_LEVEL_ORGANIZATIONS),
-        generateHasFileAggregation()
+        generateAssociatedArtifactsAggregation(),
+        generateEntityHierarchyAggregation(),
+        generateFundingSourceAggregation(),
+        generateTopLevelOrganisationAggregation()
     );
 
+    public static NestedAggregationBuilder generateAssociatedArtifactsAggregation() {
+        return
+            splittedBranchBuilder(ASSOCIATED_ARTIFACTS, ASSOCIATED_ARTIFACTS)
+                .subAggregation(branchBuilder(HAS_FILE, ATTACHMENT_VISIBLE_FOR_NON_OWNER))
+                .subAggregation(branchBuilder(LICENSE, ASSOCIATED_ARTIFACTS, LICENSE, KEYWORD)
+                );
+    }
+
+    public static AbstractAggregationBuilder<?> generateFundingSourceAggregation() {
+        return
+            splittedBranchBuilder(FUNDINGS, FUNDINGS)
+                .subAggregation(
+                    branchBuilder(ID, FUNDINGS, SOURCE, IDENTIFIER, KEYWORD)
+                        .subAggregation(
+                            generateLabelsAggregation(jsonPath(FUNDINGS, SOURCE))
+                        )
+                );
+    }
+
+    public static AbstractAggregationBuilder<?> generateEntityHierarchyAggregation() {
+        return splittedBranchBuilder(ENTITY_DESCRIPTION, ENTITY_DESCRIPTION)
+            .subAggregation(generateContributor())
+            .subAggregation(generateReference());
+    }
+
+    public static AbstractAggregationBuilder<?> generateContributor() {
+        return splittedBranchBuilder(CONTRIBUTOR, ENTITY_DESCRIPTION, CONTRIBUTORS)
+            .subAggregation(
+                branchBuilder(ID, ENTITY_DESCRIPTION, CONTRIBUTORS, IDENTITY, ID, KEYWORD)
+                    .subAggregation(
+                        branchBuilder(NAME, ENTITY_DESCRIPTION, CONTRIBUTORS, IDENTITY, NAME, KEYWORD))
+            );
+    }
+
+    public static AbstractAggregationBuilder<?> generateReference() {
```

Comment:
generateReferenceAggregation to be consistent with naming of other aggregation methods.